java/source/flight.rst

```diff
@@ -0,0 +1,1699 @@
+            listener.onNext(new ActionType("drop", "delete a flight. Action body is a JSON encoded path."));
+            listener.onCompleted();
+        }
+
+        @Override
+        public void close() throws Exception {
+            System.out.println("Calling to close");
+            AutoCloseables.close(holders.values());
+            holders.clear();
+        }
+    }
+
+
+    // ExampleFlightServer
+
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+
+    import java.io.IOException;
+
+    public class ExampleFlightServer implements AutoCloseable {
+        private final FlightServer flightServer;
+        private final Location location;
+        private final BufferAllocator allocator;
+        private final InMemoryStore mem;
+
+        /**
+         * Constructs a new instance using Allocator for allocating buffer storage that binds
+         * to the given location.
+         */
+        public ExampleFlightServer(BufferAllocator allocator, Location location) {
+            this.allocator = allocator.newChildAllocator("flight-server", 0, Long.MAX_VALUE);
+            this.location = location;
+            this.mem = new InMemoryStore(this.allocator, location);
+            this.flightServer = FlightServer.builder(allocator, location, mem).build();
+        }
+
+        public Location getLocation() {
+            return location;
+        }
+
+        public int getPort() {
+            return this.flightServer.getPort();
+        }
+
+        public void start() throws IOException {
+            flightServer.start();
+        }
+
+        public void awaitTermination() throws InterruptedException {
+            flightServer.awaitTermination();
+        }
+
+        public InMemoryStore getStore() {
+            return mem;
+        }
+
+        @Override
+        public void close() throws Exception {
+            AutoCloseables.close(mem, flightServer, allocator);
+        }
+    }
+
+    // Creating the Server
+
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+
+    BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+    ExampleFlightServer efs = new ExampleFlightServer(allocator, Location.forGrpcInsecure("localhost", 33333));
+    efs.start();
+
+    System.out.println(efs.getLocation());
+
+.. testoutput::
+
+    Location{uri=grpc+tcp://localhost:33333}
+
+Creating the Client
+*******************
+
+Get List Actions
+----------------
+
+Validate lists actions available on the Flight service:
+
+.. testcode::
```

Comment:
Hmm, we have to repeat the entire contents of the server each time? I think that will get annoying very fast.

Reply:
We could just put all of the client calls in the first code block (with println calls to separate things) and then talk about what happens afterwards.

Reply:
Yes, all code needs to work in isolation, but it's not very helpful if every example starts with a few hundred lines of copy-pasted code since then it's hard to see what the example is trying to demonstrate. The other language cookbooks don't repeat the server definition for each example, for instance. It might be easier to show the server once and then just use normal code blocks to show the client examples afterwards.

Reply:
Adding all the code at the initial documentation and continue explaining the code.

Also added a ticket to create a custom plugin to be able to run java code cookbook but only show to our user the main code: https://issues.apache.org/jira/browse/ARROW-15703

Reply:
FWIW because we show the entire server up front initially I think down below, we can just use normal code blocks to show snippets of the server instead.

Reply:
Also I think what would help (for this, and for formatting the code) is to allow this directive to accept a file, and a line range, then it will be easier to handle long examples like this.

Reply:
Ticket was created. Lets join ideas for cookbook v2